### PR TITLE
feat: Adding ``url`` property for getting emoji's url from the emoji's icon

### LINKED ISSUE #782: [REQUEST] Adding ``url`` property for getting the ``Emoji``'s URL
### Describe the feature.

Currently, there is no way to get the emoji's URL. It can be done by manually getting the emoji's ID, then create an url from there. The PR I am about to open will add an ``url`` property for doing this.

### Code of Conduct

- [X] I agree to follow the contribution requirements.

### PR DESCRIPTION
**Pull request by B1ue-Dev**
## About

This pull request is about adding an ``url`` property under ``interactions.Emoji``. Currently, the only way to get the emoji's id is by getting the emoji's id, then create an url with ``/emojis/{emoji.id}.png`` or ``/emojis/{emoji.id}.gif`` if that emoji is an animated one. The property ``url`` will make it easier to get the emoji's URL.

## Checklist

- [x] I've ran `pre-commit` to format and lint the change(s) made.
- [x] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [x] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues).
  - (If existent): https://github.com/interactions-py/library/issues/782
- [x] I've made this pull request for/as: (check all that apply)
  - [ ] Documentation
  - [ ] Breaking change
  - [x] New feature/enhancement
  - [ ] Bugfix

### CODE REVIEW

File: interactions/api/models/message.py [hunk 1]
```diff
@@ -790,6 +790,19 @@ async def delete(
...
+    def url(self) -> Optional[str]:
+        """
+        Returns the emoji's url.
+        :return URL of the emoji
+        :rtype: str
+        """
+        if not self._client:
+            raise AttributeError("HTTPClient not found!")
+        url = f"https://cdn.discordapp.com/emojis/{self.id}"
+        url += ".gif" if self.animated is True else ".png"
```
Comment (mAxYoLo01) [inline interactions/api/models/message.py]:
```suggestion
        url += ".gif" if self.animated else ".png"
```

File: interactions/api/models/message.pyi [hunk 2]
```diff
@@ -203,6 +203,8 @@ class Emoji(DictSerializerMixin):
         guild_id: int,
         reason: Optional[str] = None,
     ) -> None: ...
+    @property
+    def icon_url(self) -> str: ...
```
Comment (mAxYoLo01) [inline interactions/api/models/message.pyi]:
```suggestion
    def url(self) -> str: ...
```

File: interactions/api/models/message.py [hunk 3]
```diff
@@ -790,6 +790,19 @@ async def delete(
             guild_id=guild_id, emoji_id=int(self.id), reason=reason
         )
 
+    @property
+    def url(self) -> Optional[str]:
```
Comment (mAxYoLo01) [inline interactions/api/models/message.py]:
```suggestion
    def url(self) -> str:
```

File: interactions/api/models/message.py (hunk 1 above)
Comment (B1ue-Dev) [reply interactions/api/models/message.py]: Why removed ``is True`` though? ``self.animated`` returns boolean.

File: interactions/api/models/message.pyi (hunk 2 above)
Comment (B1ue-Dev) [reply interactions/api/models/message.pyi]: Oh right. My bad.

File: interactions/api/models/message.py (hunk 1 above)
Comment (mAxYoLo01) [reply interactions/api/models/message.py]: It's one of Python's many magic!
`boolean = True`
`if boolean is True:` will return the same as `if boolean:`

Comment (B1ue-Dev) [reply interactions/api/models/message.py]: Ohhhhh...
Yeah, I don't expect this. Thanks for that sugggestion.

**Comment by B1ue-Dev** [comment]
OK, with the help of Max, I think this is ready to be merged.

**Comment by AstreaTSS** [comment]
Noted! Just note that this PR is on hold the bug fix PRs are merged.

File: interactions/api/models/message.py [hunk 4]
```diff
@@ -790,6 +790,19 @@ async def delete(
...
+    @property
+    def url(self) -> str:
+        """
+        Returns the emoji's url.
+        :return URL of the emoji
+        :rtype: str
+        """
+        if not self._client:
+            raise AttributeError("HTTPClient not found!")
+        url = f"https://cdn.discordapp.com/emojis/{self.id}"
```
Comment (sillyElvy) [inline interactions/api/models/message.py:801]:
```suggestion
        url = f"https://cdn.discordapp.com/emojis/{str(self.id)}"
```

**Review by sillyElvy — CHANGES_REQUESTED** [review]

File: interactions/api/models/message.py (hunk 4 above)
Comment (AstreaTSS) [reply interactions/api/models/message.py:801]: This _should_ be unneeded, actually - f-strings should do that for you, and sourcery will complain too.

Comment (sillyElvy) [reply interactions/api/models/message.py:801]: Wait snowflake has a repr?

Comment (AstreaTSS) [reply interactions/api/models/message.py:801]: It has a `__str__`, which Python will automatically use here.

Comment (sillyElvy) [reply interactions/api/models/message.py:801]: Totally forgot about that xD

File: interactions/api/models/message.py [hunk 5]
```diff
@@ -790,6 +790,19 @@ async def delete(
...
 
+    @property
+    def url(self) -> str:
+        """
+        Returns the emoji's url.
+        :return URL of the emoji
+        :rtype: str
+        """
+        if not self._client:
+            raise AttributeError("HTTPClient not found!")
```
Comment (sillyElvy) [inline interactions/api/models/message.py]: This doesn't need the client attribute to function

**Review by sillyElvy — CHANGES_REQUESTED** [review]

File: interactions/api/models/message.py (hunk 5 above)
Comment (B1ue-Dev) [reply interactions/api/models/message.py]: OK, I followed and removed the ``_client`` function.

File: interactions/api/models/message.py [hunk 6]
```diff
@@ -790,6 +790,17 @@ async def delete(
             guild_id=guild_id, emoji_id=int(self.id), reason=reason
         )
 
+    @property
+    def url(self) -> str:
+        """
+        Returns the emoji's url.
+        :return URL of the emoji
+        :rtype: str
+        """
```
Comment (Toricane) [inline interactions/api/models/message.py]:
```suggestion
        """
        Returns the emoji's URL.

        :return: URL of the emoji
        :rtype: str
        """
```

**Review by Toricane — CHANGES_REQUESTED** [review]
Small docstring suggestion

File: interactions/api/models/message.py (hunk 6 above)
Comment (B1ue-Dev) [reply interactions/api/models/message.py]: Ha, a space between them, but okay.

**Review by Toricane — APPROVED** [review]
LGTM!

**Review by sillyElvy — APPROVED** [review]

**Pull request merged**